tests/test-ble.js

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2016, Intel Corporation.
+
+console.log("Wire IO2 to IO4");
+
+var gpio = require("gpio");
+var pins = require("arduino101_pins");
+var ble = require ("ble");
+
+var total = 0;
+var passed = 0;
+function assert(actual, description) {
+    total += 1;
+    var label = "\033[1m\033[31mFAIL\033[0m";
+    if (actual === true) {
+        passed += 1;
+        label = "\033[1m\033[32mPASS\033[0m";
+    }
+    console.log(label + " - " + description);
+}
+
+function expectThrow(description, func) {
+    var threw = false;
+    try {
+        func();
+    }
+    catch (e) {
+        threw = true;
+    }
+    assert(threw, description);
+}
+
+var deviceName, bufferData, pinA, pinB, disconnectClient,
+    readValue, acceptClient, rssiValue, writeTmp;
+var writeValue = true;
+var PoweredonFlag = false;
+var advertisingFlag = false;
+var writeFlag = true;
+var readFlag = true;
+var rssiFlag = true;
+var notifyFlag = true;
+var acceptFlag = true;
+var disconnectFlag = false;
+var advertiseFlag = true;
+var totalFlag = false;
+var stopFlag = true;
+
+pinA = gpio.open({ pin: pins.IO2 });
+pinB = gpio.open({ pin: pins.IO4, direction: "in" });
+pinA.write(writeValue);
+readValue = pinB.read();
+
+bufferData = new Buffer(1);
+bufferData.writeUInt8(0);
+
+var writeCharacteristic = new ble.Characteristic({
+    uuid: 'aa00',
+    properties: ['write'],
+    descriptors: [
+        new ble.Descriptor({
+            uuid: '2901',
+            value: 'write'
+        })
+    ]
+});
+
+var keywordResult = [ true, true, true, true ];
+
+writeCharacteristic.onWriteRequest = function(data, offset, withoutResponse,
+                                              callback) {
+    if (data.toString('hex') === "01") {
+        writeValue = true;
+    } else if (data.toString('hex') === "00") {
+        writeValue = false;
+    } else {
+        writeTmp = writeValue;
```

Comment:
maybe rename this to tmpValue or just tmp, writeTmp makes me think of temperature values.